backends/ignite_test.go

```diff
@@ -0,0 +1,636 @@
+				err:  errors.New("net/http: nil Context"),
+			},
+		},
+		{
+			desc: "http.Client.Do() returns an error",
+			in: testInput{
+				ctx:      context.TODO(),
+				httpResp: nil,
+				httpErr: &url.Error{
+					Op:  "GET",
+					Err: errors.New("fake http.Client error"),
+				},
+			},
+			expected: testOutput{
+				resp: nil,
+				err:  &url.Error{Op: "GET", Err: errors.New("fake http.Client error")},
+			},
+		},
+		{
+			desc: "nil http.Response.Body is returned",
+			in: testInput{
+				ctx: context.TODO(),
+				httpResp: &http.Response{
+					StatusCode: http.StatusOK,
+					Body:       nil,
+				},
+				httpErr: nil,
+			},
+			expected: testOutput{
+				resp: nil,
+				err:  errors.New("Ignite error. Received empty httpResp.Body"),
+			},
+		},
+		{
+			desc: "Non 200 status code is returned in the http.Response",
+			in: testInput{
+				ctx: context.TODO(),
+				httpResp: &http.Response{
+					StatusCode: http.StatusNotFound,
+					Body: fakeReadCloser{
+						body: []byte{0x0},
+						err:  io.EOF,
+					},
+				},
+				httpErr: nil,
+			},
+			expected: testOutput{
+				resp: []byte{0x0},
+				err:  errors.New("Ignite error. Unexpected status code: 404"),
+			},
+		},
+		{
+			desc: "http.Response.Body read error",
+			in: testInput{
+				ctx: context.TODO(),
+				httpResp: &http.Response{
+					StatusCode: http.StatusOK,
+					Body: fakeReadCloser{
+						body: []byte{0x0},
+						err:  io.ErrShortBuffer,
+					},
+				},
+				httpErr: nil,
+			},
+			expected: testOutput{
+				resp: nil,
+				err:  errors.New("Ignite error. IO reader error: short buffer"),
+			},
+		},
+		{
+			desc: "Success",
+			in: testInput{
+				ctx: context.TODO(),
+				headers: http.Header{
+					"HEADER": []string{"value"},
+				},
+				httpResp: &http.Response{
+					StatusCode: http.StatusOK,
+					Body: fakeReadCloser{
+						body: []byte(`{"jsonObject":"value"}`),
+						err:  io.EOF,
+					},
+				},
+				httpErr: nil,
+			},
+			expected: testOutput{
+				resp: []byte(`{"jsonObject":"value"}`),
+				err:  nil,
+			},
+		},
+	}
+	for _, tc := range testCases {
+		fakeIgniteClient := &igniteSender{
+			httpClient: &fakeHttpClient{
+				mockFunction: func() (*http.Response, error) {
+					return tc.in.httpResp, tc.in.httpErr
+				},
+			},
+		}
+		actualResp, actualErr := fakeIgniteClient.DoRequest(tc.in.ctx, &url.URL{}, tc.in.headers)
+
+		assert.Equal(t, tc.expected.resp, actualResp, tc.desc)
+		assert.Equal(t, tc.expected.err, actualErr, tc.desc)
+	}
+}
+
+func TestNewIgniteBackend(t *testing.T) {
+	type logEntry struct {
+		msg string
+		lvl logrus.Level
+	}
+
+	type testOut struct {
+		backend      *IgniteBackend
+		panicHappens bool
+		logEntries   []logEntry
+	}
+
+	type testCase struct {
+		desc     string
+		in       config.Ignite
+		expected testOut
+	}
+	testGroups := []struct {
+		desc      string
+		testCases []testCase
+	}{
+		{
+			desc: "config validation error",
+			testCases: []testCase{
+				{
+					desc: "empty scheme",
```

Comment:
Nitpick: for the "empty scheme", "empty host" and "empty port" tests, I think it would be best if all of the values were valid except for a single field so we can be sure the field under test is what is causing the error. The error message is the same for all of these empty fields so it is not possible to discern what is causing the failure as the test is currently constructed.

Reply:
done